src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -3618,6 +3627,29 @@ class RGWRadosPutObj : public RGWHTTPStreamRWRequest::ReceiveCB
       }
     }
 
+    // replication filter if crypt mode is AES256
+    if (auto i = src_attrs.find(RGW_ATTR_CRYPT_MODE); i != src_attrs.end() && i->second.to_str() == "AES256") {
+      // we need sal::Bucket of the destination bucket so we can store the attrs related to enc if needed
+      std::unique_ptr<rgw::sal::Bucket> bucket;
+      ret = driver->load_bucket(dpp, obj.bucket, &bucket, y);
+      if (ret < 0) {
+        ldpp_dout(dpp, 0) << "ERROR: failed to load bucket: " << obj.bucket << " ret= " << ret << dendl;
+        return ret;
+      }
+
+      std::unique_ptr<BlockCrypt> block_crypt;
+      std::string err_msg;
+      ret = handle_sse_s3_encryption(dpp, cct, y, bucket.get(), obj.key, &err_msg, src_attrs, &block_crypt, nullptr);
```

Comment:
Experiencing a deadlock here while it tries to create a key for the destination bucket here: https://github.com/ceph/ceph/blob/db43ab3df9421eac7676101b0fc65e6e20b9fb8e/src/rgw/rgw_kms.cc#L309

Reply:
Thanks @cbodley for finding the issue.